x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engines/engines_list_logic.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { flashAPIErrors } from '../../../shared/flash_messages';
+
+import { mockedEngines } from './_mocks_/mocked_engines';
+import { EngineListDetails, EnginesListAPIResponse } from './types';
+
+interface EngineListValues {
+  engines: EngineListDetails[];
+}
+
+interface EnginesListActions {
+  onEnginesPagination(page: number): { page: number };
+  onEnginesLoad({ results, meta }: EnginesListAPIResponse): EnginesListAPIResponse;
+  loadEngines(): void;
+}
+
+export const EnginesListLogic = kea<MakeLogicType<EngineListValues, EnginesListActions>>({
+  path: ['enterprise_search', 'engines_logic'],
+  actions: {
+    onEnginesLoad: ({ results, meta }) => ({ results, meta }),
+    loadEngines: true,
+  },
+  reducers: {
+    engines: [
+      [],
+      {
+        onEnginesLoad: (_, { results }) => results,
+      },
+    ],
+  },
+  selectors: {},
+  listeners: ({ actions }) => ({
+    loadEngines: () => {
```

Comment:
suggestion: We've moved to more modern approaches to Kea logic and API calls. In general, we want to remove the direct API calls from places like this, and instead only react to the actions emitted by those API calls. What you're doing now is the old way we did things, but that led to a lot of issues with code duplication and divergence. 

That means:
1. you need to import the `apiError` action from your apiLogic, and listen to that error to trigger the error flash.
2. You don't need to set the value of the response on this logic,  it's already available directly from your apiLogic under the `data` key. If you want to access that data in this logic, you should import the `data` key here (optionally renaming it). If you want to transform that response somehow, you should use a selector instead of a reducer.

If you look at any of the logic files in the new `enterprise_search_content/` area that is *not* crawler-logic, you'll see some examples of how we do this. 

Also, this guide to our approach to KEA may be useful: https://github.com/elastic/kibana/blob/main/x-pack/plugins/enterprise_search/KEA.md

If you need some help getting through this stuff, I'd be happy to pair :)